test/test_eq.ml

```diff
@@ -0,0 +1,108 @@
+let chr = Char.unsafe_chr
+let code = Char.code
+let unsafe_to_string = Bytes.unsafe_to_string
+
+module StdBytes = Bytes
+
+let std_invalid_arg s = invalid_arg s
+
+let list_init n f =
+  let rec go acc = function
+    | 0 -> List.rev acc
+    | n -> go (f n :: acc) (pred n) in
+  go [] n
+
+open Core_bench
+open Core.Std [@@warning "-3"]
+open Core
+
+let invalid_arg fmt = Format.ksprintf (fun s -> std_invalid_arg s) fmt
+
+let () = Random.self_init ()
+let (<.>) f g = fun x -> f (g x)
+
+let random_hash digest_size _ =
+  let rs = Bytes.create (digest_size * 2) in
+  let rec fill i =
+    if i = digest_size * 2
+    then unsafe_to_string rs
+    else match Random.int 16 with
+      | n when n < 10 ->
+        StdBytes.set rs i (chr (code '0' + n))
+      ; fill (succ i)
+      | n ->
+        StdBytes.set rs i (chr (code 'a' + n - 10))
+      ; fill (succ i) in
+  fill 0
+
+let hashes = Array.init 1000 (Digestif.SHA1.consistent_of_hex <.> random_hash Digestif.SHA1.digest_size)
+
+let bench n =
+  Bench.Test.create_indexed
+    ~name:"eq"
+    ~args:(list_init n (fun _ -> Random.int 1000))
+  @@ fun i ->
+  let a = Array.get hashes i in
+  let b = Array.get hashes i in
```

Comment:
`a` is always equal to `b` here?

Reply:
Hmmhmm, we does not care about inputs (and it's why it's random inputs) and only time needed to execute `eq` is relevant here.

Reply:
but if the input are identical, that means that eq will scan the full bytes anyway.

Reply:
The implementation should scan full bytes anyway even if `a` and `b` differs. Then I can generate 2 inputs and execute `eq` on them.